### ISSUE TITLE
Join on closest value

### ISSUE THREAD

**Issue by MaxHalford** [pandas-dev/pandas#14529]
Hello,

I am not very familiar with the current discussions, but I was wondering if there was a feature in the pipe to _**join two dataframes on the closest value of a given column**_. It happens quite often that two dataframes contain different data with different timestamps, yet it feels natural to want to join the two dataframes by matching each row on the left dataframe with the *closest* row in the right dataframe.

If you want me to add more detail please do tell me.

Regards.

**Comment by TomAugspurger** [comment]
I think you're looking for [`merge_asof`](http://pandas-docs.github.io/pandas-docs-travis/generated/pandas.merge_asof.html), does that look right?

**Comment by MaxHalford** [comment]
I hadn't noticed that method... Beautiful :).

Sorry for bothering you, have a nice day!

**Comment by MaxHalford** [comment]
Actually I was a bit hasty, I just realized that this isn't exactly what I want. `merge_asof` matches to the **upper closest**, not to the closest.

Do you see what I mean?

**Comment by jreback** [comment]
http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.reindex.html?highlight=reindex#pandas.DataFrame.reindex

use ``tolerance`` that works in both directions.

**Comment by MaxHalford** [comment]
It doesn't work for me, the tolerance in the `as_of` function is working but it still only considers forward values, not backward ones.

**Comment by jreback** [comment]
this is on ``.reindex``

**Comment by jreback** [comment]
docs are [here](http://pandas.pydata.org/pandas-docs/stable/basics.html#limits-on-filling-while-reindexing). please post an example if you are still having difficulty.

**Comment by MaxHalford** [comment]
Sorry I wasn't available this weekend. I'm not sure I've expressed myself clearly. The issue is the following: I have two dataframes: the first one contains updates for a bike station, the second one contains weather updates. I would like to add to each station update the closest weather update.

![screen shot 2016-10-31 at 11 09 08](https://cloud.githubusercontent.com/assets/8095957/19850803/c859883c-9f5a-11e6-9595-0618175e1737.png)

![screen shot 2016-10-31 at 11 07 12](https://cloud.githubusercontent.com/assets/8095957/19850805/cccbb548-9f5a-11e6-852d-b7039f71a637.png)

I did the following:

```python
weather_df = get_weather_updates(city_name)
weather_df.sort_values(by='moment', inplace=True)
joined_df = pd.merge_asof(left=city_df, right=weather_df, on='moment')
```

But I got 

![screen shot 2016-10-31 at 11 13 42](https://cloud.githubusercontent.com/assets/8095957/19850867/202b7a48-9f5b-11e6-87f0-0dcf2afde2cd.png)

I'm not sure how I can use the reindex method because the resulting index might not contain unique values, ie. some station updates could be merged with the same weather update.
